src/components/options/settings/importExport.svelte

```diff
@@ -0,0 +1,201 @@
+
+  reader.onload = async function (event) {
+    await saveSearches(event.target.result);
+  };
+  reader.readAsText(selectedFile);
+
+  // Reset the value.
+  event.target.value = "";
+}
+
+function importFromFile() {
+  document.getElementById("settings_fileInput").click();
+}
+
+async function openModal() {
+  // Update textarea.
+  await updateJSONTextarea();
+  editModal = new BSN.Modal("#settings_editModal");
+  editModal.show();
+}
+
+async function saveModalChanges() {
+  const changes = document.getElementById("settings_json").value;
+  const success = await saveSearches(changes);
+
+  // Close modal on success.
+  if (success) {
+    closeModal();
+  }
+}
+
+async function saveSearches(data) {
+  try {
+    const parsedData = JSON.parse(data);
+
+    if (
+      confirm(
+        "Are you sure you want to override your local settings with these values?"
+      )
+    ) {
+      await ConfigFile.parseJSONFile(parsedData, true);
+
+      // Update UI according to this change.
+      dispatch("updateMainConfiguration");
+
+      Notiflix.Notify.Success("New configuration saved");
+      return true;
+    }
+  } catch (err) {
+    console.error(err);
+    Notiflix.Notify.Failure(
+      "Configuration couldn't be saved. Please verify that the file is valid."
+    );
+    return false;
+  }
+}
+
+async function updateJSONTextarea(newSettings) {
+  const textarea = document.getElementById("settings_json");
+  if (textarea) {
+    if (_.isNil(newSettings)) {
+      newSettings = await ConfigFile.generateJSONFile();
+    }
+
+    textarea.value = beautify(JSON.stringify(newSettings), {
+      indent_size: 2,
+    });
+  }
+}
+</script>
+
+<div class="row">
+  <div class="col-md-10 offset-md-1">
+    <h2 class="text-center mt-3">Export/Import Search Options</h2>
+
+    <p>
+      Use Import and Export to share settings using a file. Otherwise, click
+      Edit to modify the settings manually. This requires knowing JSON
+      formatting well. We also recommend exporting a copy of your settings prior
+      to doing this.
+    </p>
+
+    <input
+      class="d-none"
+      id="settings_fileInput"
+      on:change="{fileImported}"
+      type="file" />
+
+    <div class="text-right">
+      <button
+        type="button"
+        class="btn btn-primary mr-2"
+        on:click="{importFromFile}"
+        id="settings_import">
+        <i class="fas fa-file-import" aria-hidden="true"></i> Import
+      </button>
+      <button
+        type="button"
+        class="btn btn-primary mr-2"
+        on:click="{exportToFile}"
+        id="settings_export">
+        <i class="fas fa-file-export" aria-hidden="true"></i> Export
+      </button>
+      <button
+        type="button"
+        class="btn btn-secondary"
+        on:click="{openModal}"
+        id="settings_edit">
+        <i class="fas fa-edit" aria-hidden="true"></i> Edit manually
+      </button>
+    </div>
+  </div>
+</div>
+
+<div
+  class="modal fade"
+  id="settings_editModal"
```

Comment:
I think that now (using Sevelte) we could generate this `id` dinamically, to avoid potential  duplicated ids (e.g. `id="MODAL_ID"` and `const MODAL_ID = _.uniqueId("importExport-modal");`).